js/ag_reference.js

```diff
@@ -905,3 +921,181 @@ AttributeGroupReferenceAggregateFn.prototype = {
         return "cnt_d(" + this._ref.shortestKey[0].term + ")";
     }
 };
+
+
+module.BucketAttributeGroupReference = BucketAttributeGroupReference;
+/**
+ * @namespace ERMrest.BucketAttributeGroupReference
+ */
+
+/**
+ * Constructs a Reference object based on {@link ERMrest.AttributeGroupReference}.
+ *
+ * This object will be the main object that client will interact with, when we want
+ * to use ermrset `attributegroup` api with the bin aggregate. References are immutable
+ * and therefore can be safely passed around and used between multiple client components
+ * without risk that the underlying reference to server-side resources could change.
+ *
+ * Usage:
+ *  - Clients _do not_ directly access this constructor.
+ *  - This will currently be used by the aggregateGroup histogram function to return a
+ *    BucketAttributeGroupReference rather than a {@link ERMrest.Reference}
+ *
+ * @param       {ERMRest.AttributeGroupColumn[]} keyColumns List of columns that will be used as keys for the attributegroup request.
+ * @param       {?ERMRest.AttributeGroupColumn[]} aggregateColumns List of columns that will create the aggreagte columns list in the request.
+ * @param       {ERMRest.AttributeGroupLocation} location  The location object.
+ * @param       {ERMRest.Catalog} catalog  The catalog object.
+ * @param       {Object} options contains bucketCount, min, max
+ * @constructor
+ */
+function BucketAttributeGroupReference(keyColumns, aggregateColumns, location, catalog, options) {
+    // call the parent constructor
+    BucketAttributeGroupReference.superClass.call(this, keyColumns, aggregateColumns, location, catalog);
+
+    this._options = options;
+}
+
+// extend the prototype
+module._extends(BucketAttributeGroupReference, AttributeGroupReference);
+
+// properties to be overriden:
+BucketAttributeGroupReference.prototype.sort = function (sort) {
+    verify(true == false, "Invalid function");
+};
+
+BucketAttributeGroupReference.prototype.search = function (term) {
+    verify(true == false, "Invalid function");
+};
+
+/**
+ * @return {ERMRest.BucketAttributeGroupPage}
+ */
+BucketAttributeGroupReference.prototype.read = function () {
```

Comment:
Can you add a little explanation about the logic of read. The extra logic that you have added and also the expected format of output.